src/CamelKSchemaManager.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as vscode from 'vscode';
+import fetch from 'cross-fetch';
+
+const CAMELK_MODELINE_PREFIX = '# camel-k:';
+const CAMELK_SCHEMA_ID = 'camelk';
+const CAMELK_SCHEMA_URI_PREFIX = CAMELK_SCHEMA_ID + '://schema/';
+
+let camelkSchemaCache: string;
+
+export async function registerCamelKSchemaProvider(mainOutputChannel: vscode.OutputChannel) {
+	const yamlExtension = vscode.extensions.getExtension('redhat.vscode-yaml');
+	await retrieveSchemaAsCache(mainOutputChannel);
+	if(yamlExtension?.isActive) {
+		const yamlExtensionAPI = yamlExtension?.exports;
+		yamlExtensionAPI.registerContributor(CAMELK_SCHEMA_ID, requestYamlSchemaUriCallback, requestYamlSchemaContentCallback);
+	} else {
+		mainOutputChannel.appendLine('Yaml extension is not active. There won\'t be automatic completion/validation for Camel K files.');
+	}
+}
+
+async function retrieveSchemaAsCache(mainOutputChannel: vscode.OutputChannel) {
+	try {
+		const res = await fetch(getCamelKSchemaUrl());
+		if(res.status >= 400) {
+			mainOutputChannel.appendLine('Cannot retrieve Camel K schema. '+ res.statusText);
+			return Promise.resolve(undefined);
+		} else if(camelkSchemaCache === undefined) {
+			camelkSchemaCache = await res.text();
+			return camelkSchemaCache;
+		}
+	} catch(err) {
+		mainOutputChannel.appendLine('Cannot retrieve Camel K schema. '+ err);
+	}
+}
+
+function requestYamlSchemaUriCallback(resource: string): string | undefined {
+	const textDocument = vscode.workspace.textDocuments.find(document => {
+		return document.uri.toString() === resource;
+	});
+	if (textDocument
+		&& resource.endsWith('.yaml')
+		&& textDocument.getText().startsWith(CAMELK_MODELINE_PREFIX)) {
+			return CAMELK_SCHEMA_URI_PREFIX + resource;
+		}
+	return undefined;
```

Comment:
I think that is the default if you don't specify it, but I could be wrong.

Reply:
if you are not sure, it means that it is easier to understand when specifying it

Reply:
yeah, was just thinking loud. I am fine with the current way you handle it.